src/decision/location_selection.jl

```diff
@@ -0,0 +1,356 @@
+end
+function location_selection_frequencies(
+    rs::ResultSet,
+    iv_type::String;
+    n_loc_int::Int64=5,
+    ind_metrics::Vector{Int64}=_get_scen_ids(rs.ranks),
+)
+    selected_ranks = _get_iv_type(rs.ranks[scenarios=ind_metrics], iv_type)
+    ranks_frequencies = ranks_to_frequencies_ts(selected_ranks; n_ranks=n_loc_int)
+    loc_count = dropdims(sum(ranks_frequencies[ranks=1:n_loc_int], dims=[1, 3]), dims=3)[timesteps=1]
+
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray, iv_type::String; dims=:scenarios,agg_func=x->x)
+    summed_inverse_rank(rs::ResultSet, iv_type::String; dims=:scenarios,agg_func=x->x)
+    summed_inverse_rank(ranks::NamedDimsArray; dims=:scenarios,agg_func=x->x)
+ 
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `rs` : ADRIA result set.
+- `iv_type` : indicates intervention log to use ("seed", "shade" or "fog").
+- `dims` : Dimensions to sum over.
+- `agg_func` : function to transform result.
+
+# Returns 
+- Inverse rankings (i.e. the greater the number the higher ranked the site).
+"""
+function summed_inverse_rank(
+    ranks::NamedDimsArray,
+    iv_type::String;
+    dims::Symbol=:scenarios,
+    agg_func::Function=x -> x,
+)
+    selected_ranks = _get_iv_type(ranks, iv_type)
+    return summed_inverse_rank(selected_ranks; dims=dims,agg_func=agg_func)
+
+end
+function summed_inverse_rank(
+    rs::ResultSet, iv_type::String; dims::Symbol=:scenarios, agg_func::Function=x -> x
+)
+    selected_ranks = _get_iv_type(rs.ranks, iv_type)
+    return summed_inverse_rank(selected_ranks; dims=dims,agg_func=agg_func)
+
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray; dims::Symbol=:scenarios, agg_func::Function=x -> x
+)
+    n_ranks = maximum(ranks)
+    inv_ranks = agg_func(dropdims(sum(n_ranks .- ranks,dims=dims),dims=dims))
+    return inv_ranks
+end
+
+
+"""
+    _get_iv_type(ranks, iv_type)
+
+Get ranks for intervention based on name.
+    
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `iv_type` : indicates intervention log to use ("seed", "shade" or "fog").
+
+# Returns
+- Ranks for specified iv_type.
+"""
+function _get_iv_type(ranks::NamedDimsArray, iv_type::String)
+    iv_dict = Dict([("seed", 1), ("shade", 2)])
+    return ranks[intervention=iv_dict[iv_type]]
+end
+
+"""
+    _get_scen_ids(ranks)
+
+Get ranks for intervention based on name.
+    
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+
+# Returns
+- Ids for full scenario set in ranks.
+"""
+function _get_scen_ids(ranks::NamedDimsArray)
+    return collect(ranks.scenarios)
+end
```

Comment:
The function is so specific, is it really worthwhile having it?

Rather than the pattern:

```julia
function some_function(array; ids=_get_scen_ids(...))
end
```

the pattern adopted elsewhere in ADRIA (see scenario metrics and visualizations) is:

```julia
function some_function(array)
    # use scenario ids by default
end
function some_function(array, filter_func)
    # apply provided filter function to extract relevant scenarios
end
```

Reply:
Good point, I can change this